lib/matching/nearest_neighbour.cpp

```diff
@@ -6,13 +6,40 @@
  */
 
 #include "nearest_neighbour.h"
+#include "Eigen/Core"
+#include <limits.h>
 
 namespace MouseTrack {
 
-std::vector<long> NearestNeighbour::operator()(const std::vector<std::shared_ptr<const ClusterDescriptor>>& descriptors, const std::vector<ClusterChain>& chains) {
-	return std::vector<long>{};
+std::vector<long> NearestNeighbour::operator()(
+    const std::vector<std::shared_ptr<const ClusterDescriptor>> &descriptors,
+    const std::vector<ClusterChain> &chains) {
+  // Prepare an empty vector that will store the index of each chain
+  std::vector<long> matchIndex;
+  for (size_t j = 0; j < descriptors.size(); j++) {
+    // Initialize a maximum distance
+    double nearest = std::numeric_limits<double>::max();
+    for (size_t i = 0; i < chains.size(); i++) {
+      // In this case we are only really concerned with the last cluster. Get
+      // the frame.
+      FrameNumber lastFrameNumber = chains[i].descriptors().rbegin()->first;
+      // Get the cluster for that frame.
+      std::shared_ptr<const ClusterDescriptor> lastDescriptor =
+          chains[i].descriptors().find(lastFrameNumber)->second;
```

Comment:
`auto descriptor = chains[i].descriptors().rbegin();` and then use `descriptor->first/second`? Improves readability.

Reply:
I like it!